packages/ui/src/forms/fieldSchemasToFormState/addFieldStatePromise.ts

```diff
@@ -750,6 +750,19 @@ export const addFieldStatePromise = async (args: AddFieldStatePromiseArgs): Prom
         childPermissions = parentPermissions
       }
 
+      const pathSegments = path ? path.split('.') : []
+
+      const tabPassesCondition =
+        passesCondition && tab.admin?.condition
+          ? tab.admin.condition(fullData, data, { blockData, path: pathSegments, user: req.user })
+          : true
```

Comment:
If there is no admin.condition or passesCondition is false, it will always be true here.

```suggestion
      const tabPassesCondition =
        passesCondition && typeof tab.admin?.condition === 'function'
          ? tab.admin.condition(fullData, data, { blockData, path: pathSegments, user: req.user })
          : passesCondition
```